src/SigSpec.Core/SigSpecGenerator.cs

```diff
@@ -85,5 +109,47 @@ private async Task<SigSpecOperation> GenerateOperationAsync(Type type, MethodInf
 
             return operation;
         }
+
+        private async Task<SigSpecChannel> GenerateChannelAsync(Type type, MethodInfo method, JsonSchemaGenerator generator, SigSpecSchemaResolver resolver)
+        {
+            var channel = new SigSpecChannel();
+            channel.Description = await type.GetXmlSummaryAsync();
+
+            foreach (var arg in method.GetParameters())
+            {
+                var parameter = await generator.GenerateWithReferenceAndNullabilityAsync<SigSpecParameter>(
+                    arg.ParameterType,
+                    arg.GetCustomAttributes(),
+                    resolver,
+                    async (p, s) =>
+                    {
+                        p.Description = await arg.GetXmlDocumentationAsync();
+                    });
+
+                channel.Parameters[arg.Name] = parameter;
+            }
+
+            //if (method.ReturnType.IsGenericType)
+            //{
+            //    //var t1 = method.ReturnType.GenericParameterAttributes;
+            //    //var t2 = method.ReturnType.GenericParameterPosition;
+            //    //var t3 = method.ReturnType.GenericTypeArguments;
+            //    var t4 = method.ReturnType.GetGenericArguments();
+            //    //var t5 = method.ReturnType.GetGenericParameterConstraints();
+            //    var t6 = method.ReturnType.GetGenericTypeArguments();
+            //    var t7 = method.ReturnType.GetGenericTypeDefinition();
+            //}
+
+            channel.ReturnType = await generator.GenerateWithReferenceAndNullabilityAsync<JsonSchema4>(
+                    method.ReturnType.GetGenericArguments().First(),
```

Comment:
Use GetGenericTypeArguments().First() for more safety.

https://github.com/RSuter/NJsonSchema/blob/master/src/NJsonSchema/Infrastructure/ReflectionExtensions.cs#L142

Reply:
Done. https://github.com/RSuter/SigSpec/pull/4/commits/97f8a764c9de82fd84658fe5c002cc0d5e7939da